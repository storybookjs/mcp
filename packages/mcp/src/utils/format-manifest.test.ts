import { describe, it, expect } from 'vitest';
import {
	formatComponentManifest,
	formatComponentManifestMapToList,
} from './format-manifest';
import type { ComponentManifest, ComponentManifestMap } from '../types';
import fullManifestFixture from '../../fixtures/full-manifest.fixture.json' with { type: 'json' };

describe('formatComponentManifest', () => {
	it('formats all full fixtures', () => {
		expect(
			formatComponentManifest(fullManifestFixture.components.button),
		).toMatchSnapshot();
		expect(
			formatComponentManifest(fullManifestFixture.components.card),
		).toMatchSnapshot();
		expect(
			formatComponentManifest(fullManifestFixture.components.input),
		).toMatchSnapshot();
	});

	describe('component name', () => {
		it('should include component name in component_name tag', () => {
			const manifest: ComponentManifest = {
				id: 'test-component',
				path: 'src/components/TestComponent.tsx',
				name: 'TestComponent',
			};

			const result = formatComponentManifest(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<component>
				<id>test-component</id>
				<name>TestComponent</name>
				</component>"
			`);
		});
	});

	describe('description section', () => {
		it('should include description when provided', () => {
			const manifest: ComponentManifest = {
				id: 'button',
				path: 'src/components/Button.tsx',
				name: 'Button',
				description: 'A simple button component',
			};

			const result = formatComponentManifest(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<component>
				<id>button</id>
				<name>Button</name>
				<description>
				A simple button component
				</description>
				</component>"
			`);
		});

		it('should handle multi-line descriptions', () => {
			const manifest: ComponentManifest = {
				id: 'button',
				path: 'src/components/Button.tsx',
				name: 'Button',
				description:
					'A versatile button component.\n\nSupports multiple variants and sizes.',
			};

			const result = formatComponentManifest(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<component>
				<id>button</id>
				<name>Button</name>
				<description>
				A versatile button component.

				Supports multiple variants and sizes.
				</description>
				</component>"
			`);
		});

		it('should omit description section when not provided', () => {
			const manifest: ComponentManifest = {
				id: 'button',
				name: 'Button',
				path: 'src/components/Button.tsx',
			};

			const result = formatComponentManifest(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<component>
				<id>button</id>
				<name>Button</name>
				</component>"
			`);
		});
	});

	describe('stories section', () => {
		it('should format a single story', () => {
			const manifest: ComponentManifest = {
				id: 'button',
				name: 'Button',
				path: 'src/components/Button.tsx',
				import: 'import { Button } from "@/components";',
				stories: [
					{
						name: 'Primary',
						description: 'A primary button variant',
						snippet: '<Button variant="primary">Click me</Button>',
					},
				],
			};

			const result = formatComponentManifest(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<component>
				<id>button</id>
				<name>Button</name>
				<story>
				<story_name>Primary</story_name>
				<story_description>
				A primary button variant
				</story_description>
				<story_code>
				import { Button } from "@/components";

				<Button variant="primary">Click me</Button>
				</story_code>
				</story>
				</component>"
			`);
		});

		it('should format multiple stories', () => {
			const manifest: ComponentManifest = {
				id: 'button',
				name: 'Button',
				path: 'src/components/Button.tsx',
				import: 'import { Button } from "@/components";',
				stories: [
					{
						name: 'Primary',
						snippet: '<Button variant="primary">Primary</Button>',
					},
					{
						name: 'Secondary',
						snippet: '<Button variant="secondary">Secondary</Button>',
					},
				],
			};

			const result = formatComponentManifest(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<component>
				<id>button</id>
				<name>Button</name>
				<story>
				<story_name>Primary</story_name>
				<story_code>
				import { Button } from "@/components";

				<Button variant="primary">Primary</Button>
				</story_code>
				</story>
				<story>
				<story_name>Secondary</story_name>
				<story_code>
				import { Button } from "@/components";

				<Button variant="secondary">Secondary</Button>
				</story_code>
				</story>
				</component>"
			`);
		});

		it('should format PascalCase story names correctly', () => {
			const manifest: ComponentManifest = {
				id: 'button',
				name: 'Button',
				path: 'src/components/Button.tsx',
				stories: [
					{
						name: 'WithIcon',
						snippet: '<Button icon={<Icon />}>Click me</Button>',
					},
					{
						name: 'DisabledState',
						snippet: '<Button disabled>Disabled</Button>',
					},
				],
			};

			const result = formatComponentManifest(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<component>
				<id>button</id>
				<name>Button</name>
				<story>
				<story_name>With Icon</story_name>
				<story_code>
				<Button icon={<Icon />}>Click me</Button>
				</story_code>
				</story>
				<story>
				<story_name>Disabled State</story_name>
				<story_code>
				<Button disabled>Disabled</Button>
				</story_code>
				</story>
				</component>"
			`);
		});

		it('should use story import over component import when provided', () => {
			const manifest: ComponentManifest = {
				id: 'button',
				name: 'Button',
				path: 'src/components/Button.tsx',
				import: 'import { Button } from "@/components";',
				stories: [
					{
						name: 'WithCustomImport',
						import: 'import { Button } from "@/custom-path";',
						snippet: '<Button>Custom</Button>',
					},
				],
			};

			const result = formatComponentManifest(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<component>
				<id>button</id>
				<name>Button</name>
				<story>
				<story_name>With Custom Import</story_name>
				<story_code>
				import { Button } from "@/custom-path";

				<Button>Custom</Button>
				</story_code>
				</story>
				</component>"
			`);
		});

		it('should handle stories without description', () => {
			const manifest: ComponentManifest = {
				id: 'button',
				name: 'Button',
				path: 'src/components/Button.tsx',
				stories: [
					{
						name: 'Simple',
						snippet: '<Button>Simple</Button>',
					},
				],
			};

			const result = formatComponentManifest(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<component>
				<id>button</id>
				<name>Button</name>
				<story>
				<story_name>Simple</story_name>
				<story_code>
				<Button>Simple</Button>
				</story_code>
				</story>
				</component>"
			`);
		});

		it('should handle stories without import', () => {
			const manifest: ComponentManifest = {
				id: 'button',
				name: 'Button',
				path: 'src/components/Button.tsx',
				stories: [
					{
						name: 'NoImport',
						snippet: '<Button>No Import</Button>',
					},
				],
			};

			const result = formatComponentManifest(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<component>
				<id>button</id>
				<name>Button</name>
				<story>
				<story_name>No Import</story_name>
				<story_code>
				<Button>No Import</Button>
				</story_code>
				</story>
				</component>"
			`);
		});

		it('should omit stories when no stories are provided', () => {
			const manifest: ComponentManifest = {
				id: 'button',
				name: 'Button',
				path: 'src/components/Button.tsx',
				description: 'A button component',
			};

			const result = formatComponentManifest(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<component>
				<id>button</id>
				<name>Button</name>
				<description>
				A button component
				</description>
				</component>"
			`);
		});

		it('should omit stories when stories array is empty', () => {
			const manifest: ComponentManifest = {
				id: 'button',
				name: 'Button',
				path: 'src/components/Button.tsx',
				stories: [],
			};

			const result = formatComponentManifest(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<component>
				<id>button</id>
				<name>Button</name>
				</component>"
			`);
		});

		it('should skip examples without snippets (e.g., when there are errors)', () => {
			const manifest: ComponentManifest = {
				id: 'button',
				name: 'Button',
				path: 'src/components/Button.tsx',
				import: 'import { Button } from "@/components";',
				examples: [
					{
						name: 'BrokenExample',
						description: 'This example has an error and no snippet',
						// No snippet property - should be skipped
					},
					{
						name: 'WorkingExample',
						description: 'This example works fine',
						snippet: '<Button>Click me</Button>',
					},
				],
			};

			const result = formatComponentManifest(manifest);

			// Should only include the working example, not the broken one
			expect(result).toMatchInlineSnapshot(`
				"<component>
				<id>button</id>
				<name>Button</name>
				<example>
				<example_name>Working Example</example_name>
				<example_description>
				This example works fine
				</example_description>
				<example_code>
				import { Button } from "@/components";

				<Button>Click me</Button>
				</example_code>
				</example>
				</component>"
			`);
		});
	});

	describe('complete component', () => {
		it('should format a complete component with description and multiple stories', () => {
			const manifest: ComponentManifest = {
				id: 'button',
				name: 'Button',
				path: 'src/components/Button.tsx',
				description:
					'A versatile button component.\n\nSupports multiple variants, sizes, and states.',
				summary: 'A button for user interactions',
				import: 'import { Button } from "@storybook/design-system";',
				stories: [
					{
						name: 'Primary',
						description: 'The primary button variant.',
						snippet:
							'const Primary = () => <Button variant="primary">Click Me</Button>',
					},
					{
						name: 'WithSizes',
						description: 'Buttons in different sizes.',
						snippet:
							'const Sizes = () => (\n  <>\n    <Button size="small">Small</Button>\n    <Button size="large">Large</Button>\n  </>\n)',
					},
				],
			};

			const result = formatComponentManifest(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<component>
				<id>button</id>
				<name>Button</name>
				<description>
				A versatile button component.

				Supports multiple variants, sizes, and states.
				</description>
				<story>
				<story_name>Primary</story_name>
				<story_description>
				The primary button variant.
				</story_description>
				<story_code>
				import { Button } from "@storybook/design-system";

				const Primary = () => <Button variant="primary">Click Me</Button>
				</story_code>
				</story>
				<story>
				<story_name>With Sizes</story_name>
				<story_description>
				Buttons in different sizes.
				</story_description>
				<story_code>
				import { Button } from "@storybook/design-system";

				const Sizes = () => (
				  <>
				    <Button size="small">Small</Button>
				    <Button size="large">Large</Button>
				  </>
				)
				</story_code>
				</story>
				</component>"
			`);
		});
	});

	describe('props section', () => {
		it('should format props from reactDocgen', () => {
			const manifest: ComponentManifest = {
				id: 'button',
				name: 'Button',
				path: 'src/components/Button.tsx',
				reactDocgen: {
					props: {
						variant: {
							description: 'The visual style variant',
							required: false,
							defaultValue: { value: '"primary"', computed: false },
							tsType: {
								name: 'union',
								raw: '"primary" | "secondary"',
								elements: [
									{ name: 'literal', value: '"primary"' },
									{ name: 'literal', value: '"secondary"' },
								],
							},
						},
						disabled: {
							description: 'Whether the button is disabled',
							required: false,
							defaultValue: { value: 'false', computed: false },
							tsType: {
								name: 'boolean',
							},
						},
						onClick: {
							description: 'Click handler',
							required: true,
							tsType: {
								name: 'signature',
								type: 'function',
								signature: {
									arguments: [{ name: 'event', type: { name: 'MouseEvent' } }],
									return: { name: 'void' },
								},
							},
						},
					},
				},
			};

			const result = formatComponentManifest(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<component>
				<id>button</id>
				<name>Button</name>
				<props>
				<prop>
				<prop_name>variant</prop_name>
				<prop_description>
				The visual style variant
				</prop_description>
				<prop_type>"primary" | "secondary"</prop_type>
				<prop_required>false</prop_required>
				<prop_default>"primary"</prop_default>
				</prop>
				<prop>
				<prop_name>disabled</prop_name>
				<prop_description>
				Whether the button is disabled
				</prop_description>
				<prop_type>boolean</prop_type>
				<prop_required>false</prop_required>
				<prop_default>false</prop_default>
				</prop>
				<prop>
				<prop_name>onClick</prop_name>
				<prop_description>
				Click handler
				</prop_description>
				<prop_type>(event: MouseEvent) => void</prop_type>
				<prop_required>true</prop_required>
				</prop>
				</props>
				</component>"
			`);
		});

		it('should handle props with minimal information', () => {
			const manifest: ComponentManifest = {
				id: 'button',
				name: 'Button',
				path: 'src/components/Button.tsx',
				reactDocgen: {
					props: {
						children: {
							tsType: {
								name: 'string',
							},
						},
					},
				},
			};

			const result = formatComponentManifest(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<component>
				<id>button</id>
				<name>Button</name>
				<props>
				<prop>
				<prop_name>children</prop_name>
				<prop_type>string</prop_type>
				</prop>
				</props>
				</component>"
			`);
		});

		it('should omit props section when reactDocgen is not present', () => {
			const manifest: ComponentManifest = {
				id: 'button',
				name: 'Button',
				path: 'src/components/Button.tsx',
				description: 'A button component',
			};

			const result = formatComponentManifest(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<component>
				<id>button</id>
				<name>Button</name>
				<description>
				A button component
				</description>
				</component>"
			`);
		});

		it('should omit props section when reactDocgen has no props', () => {
			const manifest: ComponentManifest = {
				id: 'button',
				name: 'Button',
				path: 'src/components/Button.tsx',
				reactDocgen: {
					props: {},
				},
			};

			const result = formatComponentManifest(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<component>
				<id>button</id>
				<name>Button</name>
				</component>"
			`);
		});
	});
});

describe('formatComponentManifestMapToList', () => {
	it('formats the full manifest fixture', () => {
		const result = formatComponentManifestMapToList(fullManifestFixture);
		expect(result).toMatchSnapshot();
	});

	describe('component list structure', () => {
		it('should format a single component', () => {
			const manifest: ComponentManifestMap = {
				v: 1,
				components: {
					button: {
						id: 'button',
						name: 'Button',
						path: 'src/components/Button.tsx',
					},
				},
			};

			const result = formatComponentManifestMapToList(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<components>
				<component>
				<id>button</id>
				<name>Button</name>
				</component>
				</components>"
			`);
		});

		it('should format multiple components', () => {
			const manifest: ComponentManifestMap = {
				v: 1,
				components: {
					button: {
						id: 'button',
						name: 'Button',
						path: 'src/components/Button.tsx',
					},
					card: {
						id: 'card',
						name: 'Card',
						path: 'src/components/Card.tsx',
					},
					input: {
						id: 'input',
						name: 'Input',
						path: 'src/components/Input.tsx',
					},
				},
			};

			const result = formatComponentManifestMapToList(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<components>
				<component>
				<id>button</id>
				<name>Button</name>
				</component>
				<component>
				<id>card</id>
				<name>Card</name>
				</component>
				<component>
				<id>input</id>
				<name>Input</name>
				</component>
				</components>"
			`);
		});
	});

	describe('summary section', () => {
		it('should include summary when provided', () => {
			const manifest: ComponentManifestMap = {
				v: 1,
				components: {
					button: {
						id: 'button',
						name: 'Button',
						path: 'src/components/Button.tsx',
						summary: 'A versatile button component',
					},
				},
			};

			const result = formatComponentManifestMapToList(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<components>
				<component>
				<id>button</id>
				<name>Button</name>
				<summary>
				A versatile button component
				</summary>
				</component>
				</components>"
			`);
		});

		it('should prefer summary over description', () => {
			const manifest: ComponentManifestMap = {
				v: 1,
				components: {
					button: {
						id: 'button',
						name: 'Button',
						path: 'src/components/Button.tsx',
						summary: 'Short summary',
						description: 'This is a longer description that should be ignored',
					},
				},
			};

			const result = formatComponentManifestMapToList(manifest);

			expect(result).toContain('Short summary');
			expect(result).not.toContain('longer description');
		});

		it('should use description when summary is not provided', () => {
			const manifest: ComponentManifestMap = {
				v: 1,
				components: {
					button: {
						id: 'button',
						name: 'Button',
						path: 'src/components/Button.tsx',
						description: 'A simple button component',
					},
				},
			};

			const result = formatComponentManifestMapToList(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<components>
				<component>
				<id>button</id>
				<name>Button</name>
				<summary>
				A simple button component
				</summary>
				</component>
				</components>"
			`);
		});

		it('should truncate long descriptions to 90 characters', () => {
			const manifest: ComponentManifestMap = {
				v: 1,
				components: {
					button: {
						id: 'button',
						name: 'Button',
						path: 'src/components/Button.tsx',
						description:
							'This is a very long description that exceeds ninety characters and should be truncated with ellipsis',
					},
				},
			};

			const result = formatComponentManifestMapToList(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<components>
				<component>
				<id>button</id>
				<name>Button</name>
				<summary>
				This is a very long description that exceeds ninety characters and should be truncated wit...
				</summary>
				</component>
				</components>"
			`);
		});

		it('should not truncate descriptions under 90 characters', () => {
			const manifest: ComponentManifestMap = {
				v: 1,
				components: {
					button: {
						id: 'button',
						name: 'Button',
						path: 'src/components/Button.tsx',
						description:
							'A description with exactly eighty characters is fine and should not be truncated',
					},
				},
			};

			const result = formatComponentManifestMapToList(manifest);

			expect(result).toContain(
				'A description with exactly eighty characters is fine and should not be truncated',
			);
			expect(result).not.toContain('...');
		});

		it('should omit summary section when neither summary nor description provided', () => {
			const manifest: ComponentManifestMap = {
				v: 1,
				components: {
					button: {
						id: 'button',
						name: 'Button',
						path: 'src/components/Button.tsx',
					},
				},
			};

			const result = formatComponentManifestMapToList(manifest);

			expect(result).not.toContain('<summary>');
			expect(result).toMatchInlineSnapshot(`
				"<components>
				<component>
				<id>button</id>
				<name>Button</name>
				</component>
				</components>"
			`);
		});
	});

	describe('complete manifest', () => {
		it('should format a complete manifest with varied components', () => {
			const manifest: ComponentManifestMap = {
				v: 1,
				components: {
					button: {
						id: 'button',
						name: 'Button',
						path: 'src/components/Button.tsx',
						summary: 'A versatile button component',
					},
					card: {
						id: 'card',
						name: 'Card',
						path: 'src/components/Card.tsx',
						description: 'A flexible container for grouping content',
					},
					input: {
						id: 'input',
						name: 'Input',
						path: 'src/components/Input.tsx',
						summary: 'Text input with validation',
						description:
							'A comprehensive input component with validation, error states, and accessibility features',
					},
					modal: {
						id: 'modal',
						name: 'Modal',
						path: 'src/components/Modal.tsx',
					},
				},
			};

			const result = formatComponentManifestMapToList(manifest);

			expect(result).toMatchInlineSnapshot(`
				"<components>
				<component>
				<id>button</id>
				<name>Button</name>
				<summary>
				A versatile button component
				</summary>
				</component>
				<component>
				<id>card</id>
				<name>Card</name>
				<summary>
				A flexible container for grouping content
				</summary>
				</component>
				<component>
				<id>input</id>
				<name>Input</name>
				<summary>
				Text input with validation
				</summary>
				</component>
				<component>
				<id>modal</id>
				<name>Modal</name>
				</component>
				</components>"
			`);
		});
	});
});
