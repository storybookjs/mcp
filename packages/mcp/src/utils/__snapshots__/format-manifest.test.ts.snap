// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`formatComponentManifest > formats all full fixtures 1`] = `
"<component>
<id>button</id>
<name>Button</name>
<description>
A versatile button component that supports multiple variants, sizes, and states.

The Button component is a fundamental building block for user interactions. It can be styled as primary, secondary, or tertiary actions, and supports disabled and loading states.

## Usage

Buttons should be used for actions that affect the current page or trigger operations. For navigation, consider using a Link component instead.
</description>
<example>
<example_name>Primary</example_name>
<example_description>
The primary button variant is used for the main call-to-action on a page. It has the highest visual prominence and should be used sparingly to guide users toward the most important action.

## Best Practices

- Use only one primary button per section
- Keep button text concise and action-oriented
- Ensure sufficient contrast for accessibility
</example_description>
<example_code>
import { Button } from '@storybook/design-system';

const Primary = () => <Button variant="primary">Click Me</Button>
</example_code>
</example>
<example>
<example_name>Secondary</example_name>
<example_description>
The secondary button variant is used for secondary actions that are still important but not the primary focus of the page.

Secondary buttons have less visual weight than primary buttons and can be used multiple times on a page.
</example_description>
<example_code>
import { Button } from '@storybook/design-system';

const Secondary = () => <Button variant="secondary">Cancel</Button>
</example_code>
</example>
<example>
<example_name>With Sizes</example_name>
<example_description>
Buttons are available in three sizes: small, medium (default), and large.

Choose the appropriate size based on the context and hierarchy of actions. Larger buttons are more prominent and easier to tap on mobile devices.
</example_description>
<example_code>
import { Button } from '@storybook/design-system';

const WithSizes = () => (
  <>
    <Button size="small">Small</Button>
    <Button size="medium">Medium</Button>
    <Button size="large">Large</Button>
  </>
)
</example_code>
</example>
<example>
<example_name>Loading</example_name>
<example_description>
The loading state provides visual feedback when an async operation is in progress.

When loading is true, the button displays a spinner and is automatically disabled to prevent multiple submissions. The button text remains visible to maintain layout stability.
</example_description>
<example_code>
import { Button } from '@storybook/design-system';

const Loading = () => <Button loading onClick={() => console.log('clicked')}>Submit</Button>
</example_code>
</example>
<example>
<example_name>Danger</example_name>
<example_description>
The danger variant is used for destructive actions that cannot be easily undone, such as deleting data or canceling subscriptions.

Use this variant to draw attention to the serious nature of the action. Consider adding a confirmation dialog for critical operations.
</example_description>
<example_code>
import { Button } from '@storybook/design-system';

const Danger = () => <Button variant="danger" onClick={() => confirm('Are you sure?')}>Delete Account</Button>
</example_code>
</example>
</component>"
`;

exports[`formatComponentManifest > formats all full fixtures 2`] = `
"<component>
<id>card</id>
<name>Card</name>
<description>
A flexible container component for grouping related content with optional header, footer, and action areas.

The Card component provides a consistent way to present information in a contained, elevated surface. It's commonly used for displaying articles, products, user profiles, or any grouped content that benefits from visual separation.

## Design Principles

- Cards should contain a single subject or action
- Maintain consistent padding and spacing
- Use elevation to indicate interactive vs static cards
- Keep content hierarchy clear with proper use of typography
</description>
<example>
<example_name>Basic</example_name>
<example_description>
A basic card with just content.

The default elevated variant provides subtle depth through shadow, making the card appear to float above the page. This is ideal for creating visual hierarchy and grouping related information.
</example_description>
<example_code>
import { Card } from '@storybook/design-system';

const Basic = () => (
  <Card>
    <h3>Card Title</h3>
    <p>This is some card content that provides information to the user.</p>
  </Card>
)
</example_code>
</example>
<example>
<example_name>With Header And Footer</example_name>
<example_description>
A card with distinct header and footer sections.

Headers typically contain titles, subtitles, or avatars. Footers often contain actions like buttons or metadata like timestamps. The header and footer are visually separated from the main content area.
</example_description>
<example_code>
import { Card } from '@storybook/design-system';

const WithHeaderAndFooter = () => (
  <Card
    header={<h3>Article Title</h3>}
    footer={<Button variant="tertiary">Read More</Button>}
  >
    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.</p>
  </Card>
)
</example_code>
</example>
<example>
<example_name>Clickable</example_name>
<example_description>
An interactive card that responds to clicks.

Clickable cards add hover effects and cursor changes to indicate interactivity. This pattern is useful for navigation cards, product cards, or any scenario where the entire card acts as a single interactive element.

## Accessibility

Clickable cards are rendered as buttons with proper keyboard support and ARIA attributes.
</example_description>
<example_code>
import { Card } from '@storybook/design-system';

const Clickable = () => (
  <Card clickable onClick={() => alert('Card clicked!')}>
    <h3>Product Name</h3>
    <p>Click anywhere on this card to view details.</p>
  </Card>
)
</example_code>
</example>
<example>
<example_name>Variants</example_name>
<example_description>
Different visual variants of the card component.

- **Elevated**: Default variant with shadow for depth
- **Outlined**: Border-only variant without shadow
- **Flat**: No border or shadow, minimal visual separation

Choose variants based on your design system and the level of emphasis needed.
</example_description>
<example_code>
import { Card } from '@storybook/design-system';

const Variants = () => (
  <>
    <Card variant="elevated">
      <p>Elevated card with shadow</p>
    </Card>
    <Card variant="outlined">
      <p>Outlined card with border</p>
    </Card>
    <Card variant="flat">
      <p>Flat card without border or shadow</p>
    </Card>
  </>
)
</example_code>
</example>
<example>
<example_name>User Profile</example_name>
<example_description>
A real-world example of a user profile card.

This example demonstrates how to compose the Card component with other design system components to create a complete, functional UI element. It includes an avatar, user information, stats, and action buttons.
</example_description>
<example_code>
import { Card, Avatar, Button } from '@storybook/design-system';

const UserProfile = () => (
  <Card
    header={
      <div style={{ display: 'flex', alignItems: 'center', gap: '1rem' }}>
        <Avatar src="/user.jpg" alt="Jane Doe" />
        <div>
          <h3>Jane Doe</h3>
          <p>Senior Developer</p>
        </div>
      </div>
    }
    footer={
      <div style={{ display: 'flex', gap: '0.5rem' }}>
        <Button variant="primary" size="small">Follow</Button>
        <Button variant="secondary" size="small">Message</Button>
      </div>
    }
  >
    <div style={{ display: 'flex', justifyContent: 'space-around' }}>
      <div><strong>1.2K</strong><br/>Followers</div>
      <div><strong>342</strong><br/>Following</div>
      <div><strong>89</strong><br/>Posts</div>
    </div>
  </Card>
)
</example_code>
</example>
</component>"
`;

exports[`formatComponentManifest > formats all full fixtures 3`] = `
"<component>
<id>input</id>
<name>Input</name>
<description>
A flexible text input component that supports various input types, validation states, and accessibility features.

The Input component is a foundational form element that wraps the native HTML input with consistent styling and behavior. It includes support for labels, error messages, helper text, and different visual states.

## Accessibility

The Input component automatically manages ARIA attributes for labels, descriptions, and error messages to ensure screen reader compatibility.
</description>
<example>
<example_name>Basic</example_name>
<example_description>
A basic text input with a label.

This is the most common use case for the Input component. Always include a label for accessibility, even if it's visually hidden in your design.
</example_description>
<example_code>
import { Input } from '@storybook/design-system';

const Basic = () => <Input label="Full Name" placeholder="Enter your name" />
</example_code>
</example>
<example>
<example_name>With Error</example_name>
<example_description>
An input displaying an error state with an error message.

Error messages should be clear, concise, and provide actionable guidance to help users fix the issue. The input border and message text are styled in red to indicate the error state.
</example_description>
<example_code>
import { Input } from '@storybook/design-system';

const WithError = () => <Input label="Email" type="email" value="invalid-email" error="Please enter a valid email address" />
</example_code>
</example>
<example>
<example_name>With Helper Text</example_name>
<example_description>
An input with helper text providing additional context or instructions.

Helper text appears below the input and provides guidance without being an error. Use it to clarify format expectations, character limits, or provide helpful hints.
</example_description>
<example_code>
import { Input } from '@storybook/design-system';

const WithHelperText = () => <Input label="Password" type="password" helperText="Must be at least 8 characters with one uppercase letter and one number" />
</example_code>
</example>
<example>
<example_name>Types</example_name>
<example_description>
Different input types for various data formats.

Using the correct input type improves the user experience by showing appropriate mobile keyboards and enabling browser validation features.
</example_description>
<example_code>
import { Input } from '@storybook/design-system';

const Types = () => (
  <>
    <Input label="Email" type="email" placeholder="user@example.com" />
    <Input label="Phone" type="tel" placeholder="(555) 123-4567" />
    <Input label="Website" type="url" placeholder="https://example.com" />
    <Input label="Age" type="number" placeholder="18" />
  </>
)
</example_code>
</example>
<example>
<example_name>Disabled</example_name>
<example_description>
A disabled input that cannot be interacted with.

Disabled inputs are useful for displaying non-editable data in forms or for inputs that become available only after certain conditions are met.
</example_description>
<example_code>
import { Input } from '@storybook/design-system';

const Disabled = () => <Input label="Username" value="john_doe" disabled />
</example_code>
</example>
</component>"
`;

exports[`formatComponentManifestMapToList > formats the full manifest fixture 1`] = `
"<components>
<component>
<id>button</id>
<name>Button</name>
<summary>
A versatile button component for user interactions
</summary>
</component>
<component>
<id>card</id>
<name>Card</name>
<summary>
A flexible container component for grouping related content
</summary>
</component>
<component>
<id>input</id>
<name>Input</name>
<summary>
A flexible text input component with validation support
</summary>
</component>
</components>"
`;
